File: v5.3/source/includes/indexes/single-field.scala
import org.mongodb.scala.{ConnectionString, MongoClient, MongoClientSettings}

// start-single-index-imports
import org.mongodb.scala._
import org.mongodb.scala.model.Indexes
import org.mongodb.scala.model.IndexOptions._
import org.mongodb.scala.model.Filters._

import scala.concurrent.Await
import scala.concurrent.duration._
import scala.util.{Failure, Success}
import java.util.concurrent.TimeUnit
// end-single-index-imports

object SingleFieldIndex {

    def main(args: Array[String]): Unit = {

    // Replace the placeholder with your Atlas connection string
    val connectionString = "<connection string>";

    // Create a new client and connect to the server
    val mongoClient = MongoClient(connectionString)
    val database = mongoClient.getDatabase("sample_mflix")
    val collection = database.getCollection("movies")

    // start-index-single
    val index = Indexes.ascending("title")
    val observable = collection.createIndex(index)
    Await.result(observable.toFuture(), Duration(10, TimeUnit.SECONDS))
    // end-index-single

    // start-index-single-query
    val filter = equal("title", "Sweethearts")

    collection.find(filter).first().subscribe((doc: Document) => println(doc.toJson()),
                            (e: Throwable) => println(s"There was an error: $e"))
    // end-index-single-query

    // start-check-single-index
    collection.listIndexes().subscribe((doc: Document) => println(doc.toJson()),
                            (e: Throwable) => println(s"There was an error: $e"))
    // end-check-single-index

    Thread.sleep(1000)
    mongoClient.close()
  }
}

File: current/source/includes/indexes/single-field.rb
require 'mongo'

# Replace the placeholders with your credentials
uri = "<connection string>"

# Sets the server_api field of the options object to Stable API version 1
options = { server_api: { version: "1" }}

# Creates a new client and connect to the server
client = Mongo::Client.new(uri, options)

# start-sample-data
database = client.use('sample_mflix')
collection = database[:movies]
# end-sample-data

# start-index-single
# Creates an index on the "title" field
collection.indexes.create_one({ title: 1 })
# end-index-single

# start-index-single-query
# Finds a document with the title "Sweethearts" by using the newly created index
filter = { title: 'Sweethearts' }
doc = collection.find(filter).first

if doc
  puts doc.to_json
else
  puts "No document found"
end
# end-index-single-query

# start-check-single-index
# Lists all indexes on the collection
puts collection.indexes.collect(&:to_json)
# end-check-single-index
